remote_execution/oss/re_grpc/src/client.rs

```diff
@@ -380,16 +338,28 @@ impl REClient {
 
     pub async fn upload(
         &self,
-        metadata: RemoteExecutionMetadata,
+        _metadata: RemoteExecutionMetadata,
         request: UploadRequest,
     ) -> anyhow::Result<UploadResponse> {
         use re_grpc_proto::build::bazel::remote::execution::v2::batch_update_blobs_request::Request;
         use re_grpc_proto::build::bazel::remote::execution::v2::compressor;
         use re_grpc_proto::build::bazel::remote::execution::v2::BatchUpdateBlobsRequest;
         use re_grpc_proto::google::rpc::Code;
 
-        let mut grpc_clients = self.grpc_clients.lock().await;
-        let client = grpc_clients.cas_client().await?;
+        let mut client = self.grpc_clients.cas_client.clone();
+
+        let files_with_digest: anyhow::Result<Vec<Request>> = request
+            .files_with_digest
+            .unwrap_or_default()
+            .into_iter()
```

Comment:
You might want to use `gazebo::prelude::*` which then gives you `into_try_map` which should simplify and avoid the need for type annotations